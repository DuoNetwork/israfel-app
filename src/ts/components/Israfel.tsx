import * as React from 'react';
import { Route, Switch } from 'react-router-dom';
import { IToken } from 'ts/common/types';
import Dex from 'ts/containers/pages/DexContainer';
import Pair from 'ts/containers/Pages/PairContainer';
import Status from 'ts/containers/Pages/StatusContainer';
import Loading from './Pages/Loading';

interface IProps {
	tokens: IToken[];
}

export default class Israfel extends React.Component<IProps> {
	public render() {
		return (
			<Switch>
					<Route path={'/status'} render={() => <Status />} />
					{this.props.tokens.map(token => (
						<Route
							key={token.code}
							path={`/${token.code}-WETH`}
							render={() => <Pair pair={`${token.code}|WETH`} />}
						/>
					))}
					{this.props.tokens.length ? (
						<Route render={() => <Dex />} />
					) : (
						<Route render={() => <Loading />} />
					)}
				</Switch>
		);
	}
}
