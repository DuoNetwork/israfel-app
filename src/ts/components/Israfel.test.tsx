/**
 * @jest-environment jsdom
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import Israfel from './Israfel';

describe('Israfel Test', () => {
	describe('User Login', () => {
		const token = [{
			address: 'test',
			code: 'test',
			denomination: 123,
			precisions: {
				['test']: 123,
			},
			feeSchedules: {
				['test']: {
					asset: 'test',
					rate: 123,
					minimum: 123
				}
			},
			maturity: 123,
		}];
		it('Test Snapshot', () => {
			const wrapper = shallow(
				<Israfel tokens={token} />
			);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
