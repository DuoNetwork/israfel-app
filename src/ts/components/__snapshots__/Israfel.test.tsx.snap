// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Israfel Test Israfel Test Test Snapshot 1`] = `
<Adapter>
  <Connect(Notification) />
  <Connect(Header) />
  <Styled(Adapter)>
    <Switch>
      <Route
        path="/status"
        render={[Function]}
      />
      <Route
        path="/pm"
        render={[Function]}
      />
      <Route
        render={[Function]}
      />
    </Switch>
  </Styled(Adapter)>
</Adapter>
`;
